### PR TITLE
Release Blocker? Disable IPv6 tests when $ZEBRA_SKIP_IPV6_TESTS is set

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Some of Zebra's new network tests fail in our `main` branch Google Cloud CI, but those failures are hidden due to bug #2403.

We need to fix this bug before we can stop hiding those failures.

## Solution

- disable IPv6 tests when $ZEBRA_SKIP_IPV6_TESTS is set
- disable more network tests when $ZEBRA_SKIP_NETWORK_TESTS is set

This allows users to disable IPv6 tests in environments where IPv6 is not configured.

## Review

@jvff reviewed PR #2277, which introduced these tests. I think this might be urgent, because we don't want to release broken code.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

## Follow Up Work

Eventually we'll want to test in IPv6-only and non-standard IPv4 localhost environments.

### CODE REVIEW

File: zebra-network/src/peer_set/initialize/tests/vectors.rs
Comment (jvff) [inline zebra-network/src/peer_set/initialize/tests/vectors.rs]: Minor: Should the environment variable names be in constants to make them easier to change and/or avoid some tests not being skipped due to possible different names?

**Review by jvff — COMMENTED** [review]
Looks good. I added one suggestion that's minor.

**Review by jvff — DISMISSED** [review]

File: zebra-network/src/peer_set/initialize/tests/vectors.rs
Comment (teor2345) [reply zebra-network/src/peer_set/initialize/tests/vectors.rs]: Yeah we can do that by adding constants to zebra_test

Comment (teor2345) [reply zebra-network/src/peer_set/initialize/tests/vectors.rs]: This is fixed in 3bc941cf958c910c9b037cab07f7aa4aef5fbc15..c987cd48c6797c4d13ad7c485b9f76390e74e56d (5 commits).

I split the search and replace commits from the manual cleanup commits, but they should be easy enough to review together.

File: zebra-test/src/net.rs
Comment (teor2345) [inline zebra-test/src/net.rs]:
```suggestion
/// Since `zebra_skip_network_tests` only disables tests which need reliable network connectivity,
/// we allow IPv6 tests even when `ZEBRA_SKIP_NETWORK_TESTS` is set.
```

File: zebrad/tests/acceptance.rs
Comment (teor2345) [inline zebrad/tests/acceptance.rs]:
```suggestion
//! ## Failures due to Configured Network Interfaces or Network Connectivity
```

**Comment by teor2345** [comment]
This PR is ready for another review.

Since I made a larger change than @jvff might have expected, I'd like another review before we merge.

**Review by jvff — APPROVED** [review]
Awesome! :tada:

**Review by jvff — APPROVED** [review]

**Pull request merged**
